components/calendar/src/ixdtf.rs

```diff
@@ -276,30 +275,31 @@ impl DateTime<AnyCalendar> {
     /// assert_eq!(datetime.time.second.number(), 17);
     /// assert_eq!(datetime.time.nanosecond.number(), 45000000);
     /// ```
-    #[cfg(feature = "compiled_data")]
-    pub fn try_from_str(ixdtf_str: &str) -> Result<Self, ParseError> {
-        Self::try_from_utf8(ixdtf_str.as_bytes())
+    pub fn try_from_str(ixdtf_str: &str, calendar: A) -> Result<Self, ParseError> {
+        Self::try_from_utf8(ixdtf_str.as_bytes(), calendar)
     }
 
     /// Creates a [`DateTime`] in any calendar from an IXDTF syntax string with compiled data.
     ///
     /// See [`Self::try_from_str()`].
-    ///
-    /// ✨ *Enabled with the `compiled_data` and `ixdtf` Cargo features.*
-    #[cfg(feature = "compiled_data")]
-    pub fn try_from_utf8(ixdtf_str: &[u8]) -> Result<Self, ParseError> {
+    pub fn try_from_utf8(ixdtf_str: &[u8], calendar: A) -> Result<Self, ParseError> {
         let ixdtf_record = IxdtfParser::from_utf8(ixdtf_str).parse()?;
+        if let Some(ixdtf_calendar) = ixdtf_record.calendar {
```

Comment:
Question/Issue: Why are you only checking the calendar if it is not None? I think we should be strict.

Reply:
If there's no calendar then the string is ISO. This is the same thing as the `Z[Europe/Zurich]` notation, which is a UTC time that we need to convert to Zurich time (just that we can actually convert calendars).